### ISSUE TITLE
WGSL validation error triggers a 'no entry at key' panic instead.

### ISSUE THREAD

**Issue by setzer22** [gfx-rs/wgpu#2381]
**Description**
I was writing some WGSL code (see: https://pastebin.com/gfqnQZZS), and instead of a validation error, I got the following panic:

```
thread 'main' panicked at 'no entry found for key', /home/josep/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.12.1/src/validation.rs:948:40
```

Here's the full backtrace:
```
  0: rust_begin_unwind
             at /rustc/532d2b14c05f9bc20b2d27cbb5f4550d28343a36/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/532d2b14c05f9bc20b2d27cbb5f4550d28343a36/library/core/src/panicking.rs:107:14
   2: core::panicking::panic_display
             at /rustc/532d2b14c05f9bc20b2d27cbb5f4550d28343a36/library/core/src/panicking.rs:63:5
   3: core::option::expect_failed
             at /rustc/532d2b14c05f9bc20b2d27cbb5f4550d28343a36/library/core/src/option.rs:1663:5
   4: core::option::Option<T>::expect
             at /rustc/532d2b14c05f9bc20b2d27cbb5f4550d28343a36/library/core/src/option.rs:710:21
   5: <std::collections::hash::map::HashMap<K,V,S> as core::ops::index::Index<&Q>>::index
             at /rustc/532d2b14c05f9bc20b2d27cbb5f4550d28343a36/library/std/src/collections/hash/map.rs:1158:9
   6: wgpu_core::validation::Interface::new
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.12.1/src/validation.rs:948:40
   7: wgpu_core::device::Device<A>::create_shader_module
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.12.1/src/device/mod.rs:1086:13
   8: wgpu_core::device::<impl wgpu_core::hub::Global<G>>::device_create_shader_module
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.12.1/src/device/mod.rs:4038:32
   9: <wgpu::backend::direct::Context as wgpu::Context>::device_create_shader_module
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.12.0/src/backend/direct.rs:1043:27
  10: wgpu::Device::create_shader_module
             at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.12.0/src/lib.rs:1705:17
```

**Repro steps**
1. Write the code I shared in a `shader.wgsl` file.
2. Attempt to compile it using
```rust
        let shader = device.create_shader_module(&wgpu::ShaderModuleDescriptor {
            label: Some("My Shader"),
            source: wgpu::ShaderSource::Wgsl(
                include_str!("./shader.wgsl").into(),
            ),
        });
```

**Expected vs observed behavior**
The validation error in my shader should be displayed. Instead, the panic is obscuring the real issue.

**Platform**
Using wgpu 0.12 under Arch Linux

**Comment by setzer22** [comment]
Thanks! But unfortunately that didn't fix it: I just tried updating to 0.12.2 (using `cargo update -p naga -p wgpu-core -p wgpu-hal`) and I can still reproduce the issue.

**Comment by cwfitzgerald** [comment]
This _feels_ like https://github.com/gfx-rs/naga/pull/1639 which hasn't hit crates yet.

**Comment by kvark** [comment]
Right! I remember fixing that. Let's make a Naga patch release then.

**Comment by kvark** [comment]
Let's close this as a dupe of #2351

**Comment by kvark** [comment]
Fix is published in naga-0.8.2

**Comment by setzer22** [comment]
Can confirm the change is fixed on 0.8.2 :+1:
